### PR TITLE
Calculate and save results in gigatonnes

### PR DESCRIPTION
**Pull request by liviajakob**
Simple changes to convert and save out the results within the glambie processing algorithm to gigatonnes.

Done on regional as well as global level, using existing conversion functions

### CODE REVIEW

File: tests/processing/test_process_global_results.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-from glambie.processing.process_global_results import _combine_regional_results_into_global
+from glambie.processing.process_global_results import _combine_regional_results_in_mwe_into_global
```
Comment (alex-ew) [inline tests/processing/test_process_global_results.py:2]: No tests for new functions. The new functions should have at least some cursory coverage.

**Review by alex-ew — CHANGES_REQUESTED** [review]
Mostly seems straight forward. The biggest issue is the duplicated function that, unless I've missed something, would be trivial to extend an existing function to handle. It would be better to some rudimentary test coverage of the added features, too.

File: glambie/processing/process_regional_results.py [hunk 2]
```diff
@@ -120,18 +122,61 @@ def combine_within_one_region(catalogue_data_group_results: DataCatalogue,
...
+    output_path_handler : OutputPathHandler
+        object to handle output path. If set to None, no plots / other data will be saved
+
+    Returns
+    -------
+    Tuple[DataCatalogue, Timeseries]
+        Unit converted data catalogue and region timeseries (in Gigatonnes)
+    """
+    # convert to gigatonnes
+    combined_region_timeseries_gt = combined_region_timeseries.apply_area_change(apply_change=False)
```
[the remark this answers is not in the document]
Comment (liviajakob) [reply glambie/processing/process_regional_results.py:159]: No it's for applying or removing the area change. I have now renamed and I think its more clear now?

Now:
apply_or_remove_area_change(apply_area_change=False)

File: glambie/processing/process_global_results.py [hunk 3]
```diff
@@ -160,3 +184,55 @@ def _combine_regional_results_into_global(regional_results_catalogue: DataCatalo
 
     return Timeseries(region=REGIONS["global"], data_group=GLAMBIE_DATA_GROUPS["consensus"],
                       data=ts_data, unit=reference_dataset_for_metadata.unit)
+
+
+def _combine_regional_results_in_gt_into_global(regional_results_catalogue: DataCatalogue) -> Timeseries:
```
[the remark this answers is not in the document]
Comment (liviajakob) [reply glambie/processing/process_global_results.py:189]: Yes, that was lazy! It's one function now!

File: glambie/processing/main.py [hunk 4]
```diff
@@ -60,28 +61,38 @@ def _run_regional_results(glambie_run_config: GlambieRunConfig,
...
+            combined_results_mwe = combine_within_one_region(results_one_region, output_path_handler)
+            combined_regional_results_mwe.append(combined_results_mwe)
+            # results in gt
+            _, combined_results_gt = convert_and_save_one_region_to_gigatonnes(
+                catalogue_data_group_results=results_one_region,
+                combined_region_timeseries=combined_results_mwe,
+                output_path_handler=output_path_handler)
+            combined_regional_results_gt.append(combined_results_gt)
+
+    # catalogue_data_group_results_per_region = DataCatalogue.from_list(
```
[the remark this answers is not in the document]
Comment (liviajakob) [reply glambie/processing/main.py:88]: done

**Comment by liviajakob** [comment]
Thanks for feedback. That's it done :)
Main change is that I combined global aggregation into 1 function and added a unit test for gigatonnes.

File: glambie/processing/process_regional_results.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alex-ew) [reply glambie/processing/process_regional_results.py]: I see. This is definitely clearer and I can see why you wouldn't want to duplicate the logic in the function.

I think part of this comes down to the fact "area_change" itself isn't immediately clear to me. adjust_area_assuming_linear_trend"? I was going to suggest something like the following but I'm not convinced it's any clearer.. Anyway, I'll leave for now

class TimeSeries:

    def apply_area_change(self, rgi_area_version: int = 6):
        return self._apply_or_remove_area_change(rgi_area_version=rgi_area_version, apply_area_change=True)

    def remove_area_change(self, rgi_area_version: int = 6):
        return self._apply_or_remove_area_change(rgi_area_version=rgi_area_version, apply_area_change=False)

    def _apply_or_remove_area_change(self, rgi_area_version: int = 6, apply_area_change: bool = True):

File: glambie/processing/process_global_results.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (alex-ew) [reply glambie/processing/process_global_results.py]: This is much cleaner. Thanks

**Review by alex-ew — APPROVED** [review]
The changes look cleaner - especially de-duplicating the combine regional results one

**Pull request merged**